src/wallet/wallet2.cpp

```diff
@@ -6221,15 +6221,51 @@ float wallet2::get_output_relatedness(const transfer_details &td0, const transfe
   return 0.0f;
 }
 //----------------------------------------------------------------------------------------------------
-size_t wallet2::pop_best_value_from(const transfer_container &transfers, std::vector<size_t> &unused_indices, const std::vector<size_t>& selected_transfers, bool smallest) const
+size_t wallet2::pop_best_value_from(const transfer_container &transfers, std::vector<size_t> &unused_indices, const std::vector<size_t>& selected_transfers, bool smallest, const std::vector<cryptonote::tx_destination_entry> &dsts) const
 {
   std::vector<size_t> candidates;
   float best_relatedness = 1.0f;
+
+  bool to_self = true;
+  for (const auto &dst: dsts)
+  {
+    if (m_subaddresses.find(dst.addr.m_spend_public_key) == m_subaddresses.end())
```

Comment:
Doesn't this only check self spends to subaddresses and doesn't cover the case where you send to your own primary addresses?